File: common/src/main/java/net/impactdev/gts/common/ui/Historical.java
package net.impactdev.gts.common.ui;

import java.util.Optional;
import java.util.function.Supplier;

pulic interface Historical<T> {

    /**
     * Represents an interface or other design which is generated via a parent of the same nature.
     * This provides a mapping
     *
     * @return
     */
    Optional<Supplier<T>> getParent();

}

File: common/src/main/kotlin/net/impactdev/gts/common/ui/Historical.kt
package net.impactdev.gts.common.ui

import java.util.*
import java.util.function.Supplier

interface Historical<T> {
    /**
     * Represents an interface or other design which is generated via a parent of the same nature.
     * This provides a mapping
     *
     * @return
     */
    val parent: Optional<Supplier<T>?>?
}